shared_model/backend/protobuf/proto_proposal_factory.hpp

```diff
@@ -0,0 +1,83 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_PROTO_PROPOSAL_FACTORY_HPP
+#define IROHA_PROTO_PROPOSAL_FACTORY_HPP
+
+#include "backend/protobuf/proposal.hpp"
+#include "interfaces/iroha_internal/proposal_factory.hpp"
+#include "interfaces/iroha_internal/unsafe_proposal_factory.hpp"
+#include "proposal.pb.h"
+
+namespace shared_model {
+  namespace proto {
+
+    template <typename Validator>
+    class ProtoProposalFactory : public interface::ProposalFactory,
+                                 public interface::UnsafeProposalFactory {
+     public:
+      FactoryResult<std::unique_ptr<interface::Proposal>> createProposal(
+          interface::types::HeightType height,
+          interface::types::TimestampType created_time,
+          const interface::types::TransactionsCollectionType &transactions)
+          override {
+        return createProposal(
```

Comment:
Should the definitions of all methods be in .hpp file?

Reply:
It is a template class, so, unfortunately, yes